### PR TITLE
[ADP-3229]  Create cardano wallet network lib

### PR DESCRIPTION
**Pull request by paolino**
- [x] Extract abstract network layer in a new sub-library
- [x] Remove abandoned StakePoolSummary type in wallet lib 

I committed a mistake in moving StakePoolSummary in primitive, forgetting to remove it from wallet lib . It's amended as necessary here.

### CODE REVIEW

File: lib/network-layer/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Revision history for node-communication
+
+## 0.1.0.0 -- YYYY-mm-dd
+
+* First version. Released on an unsuspecting world.
```
Comment (paweljakubas) [inline lib/network-layer/CHANGELOG.md]: :smiley:

Comment (paweljakubas) [inline lib/network-layer/CHANGELOG.md]:
```suggestion
* First version extracted
```

File: lib/network-layer/src/Cardano/Wallet/Network/Logging/Aggregation.hs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    let s' =
+            s{time = overCurrent (const t) (time s)}
+                { prog = overCurrent (const p) (prog s)
+                }
+    atomically $ putTMVar var $ hoistStats forgetPast s'
+    return s'
+  where
+    forgetPast (Rearview _past curr) = initRearview curr
+
+-- See NOTE [PointSlotNo]
```
Comment (paweljakubas) [inline lib/network-layer/src/Cardano/Wallet/Network/Logging/Aggregation.hs:204]: is it here ? If not then you might expand on that

File: lib/network-layer/src/MyLib.hs [hunk 3]
```diff
@@ -0,0 +1,4 @@
+module MyLib (someFunc) where
```
Comment (paweljakubas) [inline lib/network-layer/src/MyLib.hs]: this is probably not needed, correct?

**Review by paweljakubas — APPROVED** [review]

File: lib/network-layer/src/MyLib.hs (hunk 3 above)
Comment (paolino) [reply lib/network-layer/src/MyLib.hs]: right

File: lib/network-layer/src/Cardano/Wallet/Network/Logging/Aggregation.hs (hunk 2 above)
Comment (paolino) [reply lib/network-layer/src/Cardano/Wallet/Network/Logging/Aggregation.hs:204]: I copied over. As always I have a bad feeling about this notes. But thanks I didn't spot it :+1:

**Pull request merged**